### PR TITLE
Kindsys: Replace "Declaration" with "Definition"

### PR DESCRIPTION
**Pull request by sdboyer**
**What is this feature?**

This replaces the term "[kind] declaration" with "[kind] definition"

**Why do we need this feature?**

"Definition" is a better word, here. Two basic reasons:
* CRDs, which certainly are relevant to what we're doing with kinds, are "definitions."
* To the extent other languages make a distinction, "definition" is closer to the thing we're referring to (the details of a symbol) than "declaration" (the name of the thing, absent details)

### CODE REVIEW

**Comment by sdboyer** [comment]
follow-up here is replacing `DeclForGen` with just a generic type

File: kinds/gen.go [hunk 1]
```diff
@@ -31,7 +31,7 @@ func main() {
 
 	// Core kinds composite code generator. Produces all generated code in
 	// grafana/grafana that derives from core kinds.
-	coreKindsGen := codejen.JennyListWithNamer(func(decl *codegen.DeclForGen) string {
+	coreKindsGen := codejen.JennyListWithNamer(func(decl *codegen.DefForGen) string {
```
Comment (joanlopez) [inline kinds/gen.go:34]:
```suggestion
	coreKindsGen := codejen.JennyListWithNamer(func(def *codegen.DefForGen) string {
```

File: pkg/codegen/generators.go [hunk 2]
```diff
@@ -10,40 +10,41 @@ import (
...
+// TODO this type is unneeded - kindsys.Kind is sufficient.
+type DefForGen struct {
+	kindsys.SomeDef
 	lin thema.Lineage
 }
 
-// Lineage returns the [thema.Lineage] for the underlying [kindsys.SomeDecl].
-func (decl *DeclForGen) Lineage() thema.Lineage {
+// Lineage returns the [thema.Lineage] for the underlying [kindsys.SomeDef].
+func (decl *DefForGen) Lineage() thema.Lineage {
```
Comment (joanlopez) [inline pkg/codegen/generators.go:41]:
```suggestion
func (def *DefForGen) Lineage() thema.Lineage {
```

**Review by joanlopez — COMMENTED** [review]
Looks like function arg names are still pending to be renamed, I added few suggestions, but would good to review them all with the IDE (or similar).

**Review by radiohead — APPROVED** [review]

**Pull request merged**